File: lang/go/test/muse/algorithms/search/LinearSearch/LinearSearchTest.go
package main

import (
    "muse/algorithms/search/LinearSearch"
    "muse/util/SequenceBuilder"
    "muse/util/TestRunner"
)

func testLinearSearch() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackIncreasing(arr)

    if LinearSearch.Find(arr, -1) != size {
        return false
    }

    if LinearSearch.Find(arr, 2147483647) != size {
        return false
    }

    for i, v := range arr {
        if LinearSearch.Find(arr, v) != i {
            return false
        }
    }

    return true
}

func main() {
    TestRunner.ParseTest(testLinearSearch())
}

File: lang/nodejs/test/muse/algorithms/search/LinearSearchTest.js
"use strict"

const LinearSearch    = require("muse/algorithms/search/LinearSearch")
const SequenceBuilder = require("muse/util/SequenceBuilder")
const TestRunner      = require("muse/util/TestRunner")

const testLinearSearch = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packIncreasing(arr)

    if (LinearSearch.find(arr, -1) !== size) {
        return false
    }

    if (LinearSearch.find(arr, 2147483647) !== size) {
        return false
    }

    for (let i = 0; i < size; i++) {
        if (LinearSearch.find(arr, arr[i]) !== i) {
            return false
        }
    }

    return true
}

if (module === require.main) {
    TestRunner.parseTest(testLinearSearch())
}
